DESIGN.md

```diff
@@ -0,0 +1,818 @@
+  "message": "upload-speed (elapsed 9.55 s) 20996.89 kbit/s (num_streams 1)"
+}
+```
+
+where `key` identifies the experiment within the scope of the
+experiment group, `percentage` is the percentage of completion
+of the whole experiment group, and `message` is a message you
+may want to show to a user to further qualify what is happening.
+
+More specifically, `key` may be one of:
+
+- `"im.FacebookMessenger"`
+- `"im.Telegram"`
+- `"im.WhatsApp"`
+- `"middlebox.HTTPHeaderFieldManipulation"`
+- `"middlebox.HTTPInvalidRequestLine"`
+- `"performance.Dash"`
+- `"performance.NDT"`
+- `"websites.WebConnectivity"`
+
+Measurement engine messages are like:
+
+```JSON
+{
+  "fields": {
+    "type": "engine"
+  },
+  "level": "info",
+  "timestamp": "2019-10-03T16:52:56.327828+02:00",
+  "message": "starting facebook_messenger"
+}
+```
+
+They are warning, informational, or debugging messages emitted by
+the measurement engine to describe what it is doing.
+
+### Knowing the location of the home directory
+
+This command:
+
+```
+ooniprobe --batch info
+```
+
+emits this JSON:
+
+```JSON
+{
+  "fields": {
+    "path": "/home/sbs/.ooni"
+  },
+  "level": "info",
+  "timestamp": "2019-10-14T13:25:35.145144+02:00",
+  "message": "Home"
+}
+```
+
+where `message` is `"Home"` and `fields.path` contains the path to the
+currently used OONI's home directory. Of course, the output changes
+depending on the configured `OONI_HOME` environment variable value.
+
+## State and configuration directory
+
+This is `$HOME/.ooni` in all systems. The `ooniprobe --batch info`
+command can tell you the precise path on your system, subject to the
+override caused by the `OONI_HOME` environment variable.
+
+The state and configuration directory contains:
+
+- `assets`: directory where assets like GeoIP files are downloaded
+
+- `config.json`: default config file, used unless you specify
+another file using `--config <file>`
+
+- `db`: directory containing the SQLite3 database
+
+- `msmts`: directory containing the JSON measurements
+
+Running concurrent instances of the OONI CLI is unsupported. They
+may conflict attempting to download/update assets. This is the main
+reason why we recommend using another `OONI_HOME` when running the
+CLI from another program/app. We do this when running the CLI from
+OONI Desktop, and you should do the same in your use case.
+
+## Configuration file
+
+The configuration file has the following structure:
+
+```JSON
+{
+  "sharing": {
+    "include_ip": false,
+    "include_asn": true,
+    "include_country": true,
+    "upload_results": true
+  },
+  "test_settings": {
+    "websites": {
+      "enabled_categories": [],
+      "limit": 0
+    },
+    "instant_messaging": {
+      "enabled_tests": [
+        "facebook-messenger",
+        "whatsapp",
+        "telegram"
+      ]
+    },
+    "middlebox": {
+      "enabled_tests": [
+        "http-invalid-request-line",
+        "http-header-field-manipulation"
+      ]
+    }
+  }
+}
+```
```

Comment:
Some settings are missing here, namely:
```
    "send_crash_reports": true,
    "collector_url": "",
    "bouncer_url": "https://bouncer.ooni.io"
```

Reply:
Are these settings something that the desktop app can set? If so, then they are missing, otherwise, they're just stuff we're promising to have before we're sure it's stable.

Reply:
`send_crash_reports` they are currently not exposed in the desktop app. Ideally they would be exposed as settings in the desktop app as they were part of the previous version of OONI Probe.

They are implemented in probe-cli, so it's just a matter of exposing them in the UI code.

I will leave it up to you and @sarathms if you believe it's reasonable to include them or not.